### ISSUE TITLE
Merge the old SUBSYSTEM_SHIELD_GENERATOR code

### ISSUE THREAD

**Issue by Goober5000** [scp-fs2open/fs2open.github.com#538]
In the old fs2_open-unstable branch, I have an implementation of shield generators, which was originally removed because you can't add a new subsystem type without breaking pilot files.  With the new 3.7.0 pilot file code, this restriction no longer applies.

There are a few other goodies in fs2_open-unstable which ought to be investigated and merged as well.  I think WMC committed a major update of the camera code.

**Comment by wookieejedi** [comment]
I know this is old, but have there happened to have been any updates on this front?

**Comment by Goober5000** [comment]
No, no further updates.  If anyone wants to try picking this up and trying to merge it, I'd be happy to answer any questions.

**Comment by wookieejedi** [comment]
Gothca. Would the main commits be the two listed here, or are there more? 
https://github.com/scp-fs2open/fs2open.github.com/search?q=shield+generator&type=Commits

**Comment by Goober5000** [comment]
I don't remember, and that was back when we were on CVS so it's harder to track.  This will call for some detective work.  Look through those commits, and compare the changes with a checkout of the unstable branch.  (Search for the shield generator subsystem type, for starters.)

**Comment by EatThePath** [comment]
At this point in time, would it make more sense to simply have anyone who wants this implement it as a script?

**Comment by wookieejedi** [comment]
I would vote yes

**Comment by Goober5000** [comment]
Mhhh... I'd be hesitant to say that conclusively because there is some specialized subsystem type logic that I don't think scripts support yet.

And also, there are a few other features in the unstable branch not related to shield generators.  This would probably be a good candidate for another Project, à la the multiplayer code project or the bug triage project.  If @z64555 feels like setting one up...

**Comment by z64555** [comment]
What would the project be focused on?  Just the subsystem shield generator code or something more general like "Unstable Archeology"

**Comment by Goober5000** [comment]
Something more general, as there are quite a few features hidden away in that branch.

**Comment by z64555** [comment]
Done: https://github.com/scp-fs2open/fs2open.github.com/projects/13

**Comment by Goober5000** [comment]
It's still relevant, but it hasn't been high on my priority list.  Claude is mistaken, though: the shield generator subsystem type is not present in the current code.  There are commented-out excerpts in objectshield.cpp, but the `SUBSYSTEM_SHIELD_GENERATOR` type is not defined in model.h, and there is no code in modelread.cpp that recognizes a subsystem as a shield generator.
